components/recurring-contributions/UpdateOrderPopUp.js

```diff
@@ -0,0 +1,327 @@
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+import styled from 'styled-components';
+
+import { formatCurrency } from '../../lib/currency-utils';
+import { getErrorFromGraphqlException } from '../../lib/errors';
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+
+import FormattedMoneyAmount from '../FormattedMoneyAmount';
+import { Box, Flex } from '../Grid';
+import LoadingPlaceholder from '../LoadingPlaceholder';
+import StyledButton from '../StyledButton';
+import StyledHr from '../StyledHr';
+import StyledInputAmount from '../StyledInputAmount';
+import StyledRadioList from '../StyledRadioList';
+import StyledSelect from '../StyledSelect';
+import { P } from '../Text';
+
+const TierBox = styled(Flex)`
+  border-top: 1px solid ${themeGet('colors.black.300')};
+`;
+
+const messages = defineMessages({
+  cancel: {
+    id: 'actions.cancel',
+    defaultMessage: 'Cancel',
+  },
+  update: {
+    id: 'subscription.updateAmount.update.btn',
+    defaultMessage: 'Update',
+  },
+  save: {
+    id: 'save',
+    defaultMessage: 'Save',
+  },
+  updateTier: {
+    id: 'subscription.menu.updateTier',
+    defaultMessage: 'Update tier',
+  },
+});
+
+const updateOrderMutation = gqlV2/* GraphQL */ `
+  mutation updateOrderTierOrAmount($order: OrderReferenceInput!, $amount: Int, $tier: TierReferenceInput) {
+    updateOrder(order: $order, amount: $amount, tier: $tier) {
+      id
+    }
+  }
+`;
+
+const getTiersQuery = gql`
+  query UpdateOrderPopUpQuery($collectiveSlug: String!) {
+    Collective(slug: $collectiveSlug) {
+      id
+      slug
+      name
+      type
+      currency
+      settings
+      tiers {
+        id
+        name
+        slug
+        interval
+        currency
+        amount
+        minimumAmount
+        button
+        amountType
+        endsAt
+        type
+        presets
+      }
+    }
+  }
+`;
+
+const UpdateOrderPopUp = ({ setMenuState, contribution, createNotification, setShowPopup, router }) => {
+  const intl = useIntl();
+
+  // state management
+  const [loadingDefaultTier, setLoadingDefaultTier] = useState(true);
+  const [selectedTier, setSelectedTier] = useState(null);
+  const [amountOptions, setAmountOptions] = useState(null);
+  const [selectedAmountOption, setSelectedAmountOption] = useState(null);
+  const [inputAmountValue, setInputAmountValue] = useState(100);
+
+  // GraphQL mutations and queries
+  const [submitUpdateOrder, { loading: loadingUpdateOrder }] = useMutation(updateOrderMutation, {
+    context: API_V2_CONTEXT,
+  });
+  const { data } = useQuery(getTiersQuery, {
+    variables: {
+      collectiveSlug: contribution.toAccount.slug,
+    },
+  });
+
+  // Tier data wrangling
+
+  const getDefaultTier = tiers => {
+    if (contribution.tier === null) {
+      return first(tiers);
+    } else {
+      return first(tiers.filter(option => option.title.toLowerCase() === contribution.tier.name.toLowerCase()));
```

Comment:
Re: name, because the tier id attached to the 'contribution' in the card has been fetched via graphql v2 (Tier attached to an Order). Whereas when we query what tiers are available to that Collective, that is a graphql v1 query, so right now the ids wouldn't match. I'll be writing the graphql v2 capability to fetch Tiers on a Collective/Account in my next PR.